reference/enterprise-api-ref/inventory.markdown

```diff
@@ -0,0 +1,255 @@
+                                                                                                                                  ]
+                                                                                                                               }
+'
+```
+
+**Example Request Body:**
+
+
+```
+{  
+   "sort":"Host name",
+   "filter":{  
+      "Host name":{  
+         "l":"%value1%",
+         "nil":"%value2%"
+      }
+   },
+   "select":[  
+      "Host name",
+      "OS type",
+      "IPv4 addresses",
+      "CFEngine version",
+      "Ports listening"
+   ],
+   "hostContextExclude":[  
+      "policy_server"
+   ]
+}
+```
+
+**Example response:**
+
+
+```
+{
+  "data": [
+    {
+      "header": [
+        {
+          "columnName": "Host name",
+          "columnType": "STRING"
+        },
+        {
+          "columnName": "OS type",
+          "columnType": "STRING"
+        },
+        {
+          "columnName": "IPv4 addresses",
+          "columnType": "STRING"
+        },
+        {
+          "columnName": "CFEngine version",
+          "columnType": "STRING"
+        },
+        {
+          "columnName": "Ports listening",
+          "columnType": "STRING"
+        }
+      ],
+      "queryTimeMs": 30,
+      "rowCount": 2,
+      "rows": [
+        [
+          "host1.cfengine.com",
+          "linux",
+          "128.30.23.33",
+          "3.10.0",
+          "22, 25, 80, 443, 5308, 9000"
+        ],
+        [
+          "host2.cfengine.com",
+          "linux",
+          "184.45.1.75",
+          "3.10.0",
+          null
+        ]
+      ]
+    }
+  ],
+  "meta": {
+    "count": 1,
+    "page": 1,
+    "timestamp": 1496222472,
+    "total": 1
+  }
+}
+```
+
+## List of variables
+
+**URI:** https://hub.cfengine.com/api/variables-dictionary
+
+**Method:** GET
+
+Shows list of all inventory variables available in the system. 
+
+See more details: 
+* [Custom Inventory][Custom Inventory]
+
+**CURL request example**
+```
+curl -k --user admin:admin -X GET https://hub.cfengine.com/api/variables-dictionary
+```
+
+**Example response:**
+
+
+```
+[
+  {
+    "id": 2,
+    "attribute_name": "BIOS vendor",
+    "category": "Hardware",
+    "readonly": 1,
+    "type": "string",
+    "convert_function": null,
+    "keyname": "default.cfe_autorun_inventory_dmidecode.dmi[bios-vendor]"
+  },
+  {
+    "id": 3,
+    "attribute_name": "BIOS version",
+    "category": "Hardware",
+    "readonly": 1,
+    "type": "string",
+    "convert_function": null,
+    "keyname": "default.cfe_autorun_inventory_dmidecode.dmi[bios-version]"
+  }
+]
+```
+
+## Edit variable
+
+**URI:** https://hub.cfengine.com/api/variables-dictionary/:id
+
+**Method:** PATCH
+
+Set variables type (int/string..). This is needed for applying filtering in correct format.
+Only `readonly - 0` variable can be edited
```

Comment:
I think variable should be plural. "Only `readonly - 0` variables can be edited.

Reply:
and variable word itself may need replaced based on @michaelclelland and @maciejmrowiec comments from above.

Reply:
Along the same lines as above with naming. Editing variable makes me thing we are editing variable values. Edit inventory attribute was first thought to come to mind, but I dont like the word attribute here. Maybe "Edit inventory trait"?